src/main/java/org/magicdgs/readtools/tools/conversion/ReadsToDistmap.java

```diff
@@ -0,0 +1,116 @@
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+@CommandLineProgramProperties(oneLineSummary = "Converts any kind of ReadTools source to Distmap format.",
+        summary = "This tool converts SAM/BAM/CRAM/FASTQ formats into Distmap format "
+                + "(Pandey & Schlötterer 2013), including information from the barcodes (BC tag) "
+                + "in the read name (Illumina format) to allow keeping to some extend sample "
+                + "information if necessary.\n"
+                + "If the source is a SAM/BAM/CRAM file and the barcodes are encoded in the read "
+                + "name, the option --" + RTStandardArguments.USER_READ_NAME_BARCODE_NAME
+                + "should be used. If the barcode information is encoded in a different tag(s) "
+                + "the option --" + RTStandardArguments.RAW_BARCODE_SEQUENCE_TAG_NAME
+                + " should be used.\n"
+                + "Note: see the help for StandardizeReads for more information about the standard "
+                + "barcode information is handled in ReadTools and when it is useful.",
+        programGroup = ReadToolsConversionProgramGroup.class)
+public class ReadsToDistmap extends ReadToolsWalker {
+
+    // this tags are already handled in the Distmap writer
+    private final static List<String> RAW_BARCODES_FASTQ_TAGS =
+            Collections.singletonList(SAMTag.BC.name());
+
+    @Argument(fullName = RTStandardArguments.RAW_BARCODE_SEQUENCE_TAG_NAME, shortName = RTStandardArguments.RAW_BARCODE_SEQUENCE_TAG_NAME, doc = "Include the barcodes encoded in this tag(s) in the read name. Note: this is not necessary for input FASTQ files.", optional = true,
+            mutex = {RTStandardArguments.USER_READ_NAME_BARCODE_NAME})
+    public List<String> rawBarcodeTags = new ArrayList<>(RAW_BARCODES_FASTQ_TAGS);
+
+    @Argument(fullName = RTStandardArguments.USER_READ_NAME_BARCODE_NAME, shortName = RTStandardArguments.USER_READ_NAME_BARCODE_NAME, doc = "Use the barcode encoded in SAM/BAM/CRAM read names. Note: this is not necessary for input FASTQ files.", optional = true,
+            mutex = {RTStandardArguments.RAW_BARCODE_SEQUENCE_TAG_NAME})
+    public boolean useReadNameBarcode = false;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "Output in Distmap format (should be in a hadoop cluster)", optional = false)
+    public String output;
+
+    // the writer for the reads
+    private GATKReadWriter writer;
+
+    // the transformer for fixing the reads
+    private ReadTransformer transformer;
+
+    @Override
+    public void onTraversalStart() {
+        writer = new ReadWriterFactory().createDistmapWriter(output, isPaired());
+        // TODO: this logic should be changed
```

Comment:
The same as before, create the argument collection for this will be better before accepting this.